primitives/consensus/randomness-beacon/src/types.rs

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2025 by Ideal Labs, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use alloc::vec::Vec;
+use codec::{Decode, Encode};
+use serde::{Deserialize, Serialize};
+
+#[cfg(not(feature = "host-arkworks"))]
+use ark_bls12_381::G1Affine as G1AffineOpt;
+
+#[cfg(feature = "host-arkworks")]
+use sp_ark_bls12_381::G1Affine as G1AffineOpt;
+
+use ark_serialize::CanonicalDeserialize;
+
+/// A `pulse` represents the output from the beacon
+/// specifically an 'unchained' one
+#[derive(Clone, PartialEq, ::prost::Message, Serialize, Deserialize)]
+pub struct Pulse {
+	/// The round of the protocol when the signature was computed
+	#[prost(uint64, tag = "1")]
+	pub round: u64,
+	/// The interpolated threshold BLS sigs
+	#[prost(bytes = "vec", tag = "2")]
+	pub signature: ::prost::alloc::vec::Vec<u8>,
+}
+
+#[derive(Clone, Debug, PartialEq, codec::MaxEncodedLen, scale_info::TypeInfo, Encode, Decode)]
+pub struct OpaquePulse {
+	// The round of the beacon protocol
+	pub round: u64,
+	// A compressed BLS signature
+	pub signature: [u8; 48],
```

Comment:
Three forward slashes for docs
```suggestion
	/// The round of the beacon protocol
	pub round: u64,
	/// A compressed BLS signature
	pub signature: [u8; 48],
```